channels/applicationinsights-channel-js/src/Sender.ts

```diff
@@ -670,29 +683,65 @@ export class Sender extends BaseTelemetryPlugin implements IChannelControls {
                 try {
                     let onCompleteFuncs = {
                         xdrOnComplete: (xdr: IXDomainRequest, oncomplete: OnCompleteCallback,payload?: IPayloadData) => {
-                            let data = _getPayloadArr(payload);
-                            if (!data) {
+                            let payloadArr = _getPayloadArr(payload);
+                            if (!payloadArr) {
                                 return;
                             }
-                            return _xdrOnLoad(xdr, data);
+                            const responseText = _getResponseText(xdr);
+                            let statsbeat = _core.getStatsBeat();
+                            var endpointHost = urlParseUrl(_self._senderConfig.endpointUrl).hostname;
+                            if (xdr && (responseText + "" === "200" || responseText === "")) {
```

Comment:
you could just have this as `responseText == 200` as JavaScript will coerce both sides of a double equals `==` to the same type before comparison. Which would then make this a little smaller -- although I see this is how it was done for the other cases as well -- so maybe leave this for now